tcpdialer.go

```diff
@@ -160,41 +137,145 @@ type tcpDialer struct {
 	once sync.Once
 }
 
-const maxDialConcurrency = 1000
+// Dial dials the given TCP addr using tcp4.
+//
+// This function has the following additional features comparing to net.Dial:
+//
+//   * It reduces load on DNS resolver by caching resolved TCP addressed
+//     for DefaultDNSCacheDuration.
+//   * It dials all the resolved TCP addresses in round-robin manner until
+//     connection is established. This may be useful if certain addresses
+//     are temporarily unreachable.
+//   * It returns ErrDialTimeout if connection cannot be established during
+//     DefaultDialTimeout seconds. Use DialTimeout for customizing dial timeout.
+//
+// This dialer is intended for custom code wrapping before passing
+// to Client.Dial or HostClient.Dial.
+//
+// For instance, per-host counters and/or limits may be implemented
+// by such wrappers.
+//
+// The addr passed to the function must contain port. Example addr values:
+//
+//     * foobar.baz:443
+//     * foo.bar:80
+//     * aaa.com:8080
+func (d *TcpDialer) Dial(addr string) (net.Conn, error) {
```

Comment:
Oops, didn't notice — I think we should use TCPDialer instead of TcpDialer. That's a common abbreviation.

Reply:
You are right, I have fixed it.